tests/api/rest_entity_recursive/RestEntityRecursiveCest.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace rest_entity_recursive;
+
+/**
+ * Class RestEntityRecursiveCest.
+ *
+ * @package Rest
+ */
+class RestEntityRecursiveCest {
+
+  private $category;
+
+  private $article;
+
+  private $entityTypeManager;
+
+  private $redirect;
+
+  /**
+   * Configuring DB before run test.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  public function _before() {
+    $this->entityTypeManager = \Drupal::entityTypeManager();
+
+    // Creating category.
+    $this->category = $this->entityTypeManager->getStorage('taxonomy_term')->create([
+      'uid' => 1,
+      'vid' => 'content_categories',
+      'name' => 'Test category',
+    ]);
+    $this->category->save();
+
+    // Creating article.
+    $this->article = $this->entityTypeManager->getStorage('node')->create([
+      'uid' => 1,
+      'type' => 'news',
+      'title' => 'Test news',
+      'status' => 1,
+      'field_content_category' => $this->category,
+      'path' => [['alias' => '/test-node']],
+      'language' => 'und'
+    ]);
+    $this->article->save();
+
+    $this->redirect = $this->entityTypeManager->getStorage('redirect')->create([
+      'redirect_source' => 'test-redirect',
+      'redirect_redirect' => 'internal:/test-node',
+      'language' => 'und',
+      'status_code' => '301',
+    ]);
+    $this->redirect->save();
+
+  }
+
+  /**
+   * Delete created entities and return default config settings.
+   */
+  public function _after() {
+    $this->entityTypeManager->getStorage('node')->delete([$this->article]);
+    $this->entityTypeManager->getStorage('taxonomy_term')->delete([$this->category]);
+    $this->entityTypeManager->getStorage('redirect')->delete([$this->redirect]);
+
+  }
+
+  /**
+   * Checks location in redirect response.
+   *
+   * @param \ApiTester $I
+   * @group additional
+   */
+  public function testRedirectJsonRecursiveFormat(\ApiTester $I) {
+    drupal_flush_all_caches();
```

Comment:
@PavelBulat can you please add a comment explaining why we need drupal_flush_all_caches() here?